design/prosemirror.md

```
@@ -103,11 +135,74 @@ binding.initialize(); // Load/create tree, set up subscriptions
 binding.destroy();    // Clean up
 ```
 
+#### Intra-Block Character-Level Diff (Downstream)
+
+When a single block is replaced with a single block of the same type, `applyDocDiff()` tries `tryIntraBlockDiff()` before falling back to block-level replacement. This uses ProseMirror's `Fragment.findDiffStart()` / `findDiffEnd()` to find the minimal changed character range within the block, producing a precise `ReplaceStep` (e.g., "insert 5 characters at position 42").
+
+**Why this matters for cursors**: ProseMirror's `StepMap` maps positions through transaction steps. A block-level `ReplaceStep(blockStart, blockEnd, ...)` maps all positions inside `[blockStart, blockEnd]` to the **end of the replacement** (default `assoc=1`). This causes remote cursors inside the same paragraph to jump to the end of the block. The intra-block diff produces a step that only touches the changed characters, so ProseMirror correctly shifts cursors that are before/after the edit point.
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)